### DISCUSSION TITLE
Trying to set charge periods via HA automation

### DISCUSSION THREAD

**Discussion by steveh0553** [nathanmarlor/foxess_modbus#448] [Q&A] [answered]
HI, can any one help please?

Im trying to set a charge period using an HA automation, my code below doesn't work. (then automation triggers correctly but the entities dont change.

Thanks in advance.
---------------- yaml--------------
automation:
  - alias: Control FoxEss-Modbus Period 2
    trigger:
      - platform: numeric_state
        entity_id: input_number.current_rate
        below: 0
    action:
      - service: homeassistant.turn_on
        entity_id:
          - binary_sensor.time_period_2_enable_charge_from_grid
          - binary_sensor.time_period_2_enable_charge_force_charge
      - service: homeassistant.turn_off
        entity_id:
          - binary_sensor.time_period_1_enable_charge_from_grid
          - binary_sensor.time_period_1_enable_charge_force_charge
      - service: input_datetime.set_datetime
        data_template:
          entity_id: input_datetime.time_period_2_start
          time: "{{ now().strftime('%H:%M:%S') }}"
      - service: input_datetime.set_datetime
        data_template:
          entity_id: input_datetime.time_period_2_end
          time: "{{ (now() + timedelta(minutes=30)).strftime('%H:%M:%S') }}"
-------  end yaml ---------------------------

**Comment by canton7** [comment, marked as answer]
None of those entities are ones whose value you can set.

There are dedicated services for setting charge periods. Open Developer Tools - Services and search for "FoxESS"

**Reply by steveh0553** [reply]
Thank you for the response, 
I didn't know they existed.
Can I use those in my automation ?

Thanks again.

**Reply by canton7** [reply]
Yes, in the same way that you call any other service from an automation

**Reply by steveh0553** [reply]
Thanks again, I'll give it a try. :)

**Reply by steveh0553** [reply]
Thanks Anthony, that worked. :)

**Reply by canton7** [reply]
Glad to hear!

**Comment by phips** [comment]
What did that automation look like @steveh0553 ? Would you mind sharing the solution please and thank you 🙏

**Reply by steveh0553** [reply]
Hi phips, sorry I no longer have the automation set up, and dont have a copy of it laying around. :(

Sorry.

**Reply by phips** [reply]
No worries. Thanks for responding none the less, it's appreciated 😄

**Reply by FozzieUK** [reply]
@phips what are you looking to do?, i've got automations for most things, I might be able to help ?.

**Reply by phips** [reply]
Ah mega, thanks @FozzieUK ! Basically I just want adhoc control of time period 2, but not have to go in to the dev console to fettle. I’d like to put the time windows on my main solar dashboard and change them as I see fit. But, saying that, I’m also pondering an automation where I check battery level around 2pm, and if it’s looking poor I’ll enable time period 2 and forced charging to get some reserve built up before Octopus’ 4-7pm expensive rate period. Does that make sense? 🤔

**Reply by calum-mcfarlane** [reply]
There is a special custom card that the foxess Modbus devs wrote to allow setting the charge slots from your dashboard. Works beautifully, you can find it in the "Frontend" section of HACS. 

As for automations, Dave's your guy for that 😄

**Reply by phips** [reply]
@calum-mcfarlane That is **exactly** what I was after! Thank you, you're a genius!

**Reply by FozzieUK** [reply]
@phips the charge card sounds like it’s exactly what you need - just shout if you need anything else 😊

**Reply by canton7** [reply]
You do need to follow a couple more steps to set it up, see https://github.com/nathanmarlor/foxess_modbus_charge_period_card/

**Reply by canton7** [reply]
Ah, no, we've been added to hacs and I never noticed! Never mind. I'll update the readme there.

**Comment by phips** [comment]
Beautiful! 😆 
<img width="1135" alt="Screenshot 2023-12-26 at 11 51 19" src="https://github.com/nathanmarlor/foxess_modbus/assets/329022/6c4e1259-62a6-4d35-b7cb-79a16940f765">
